pkg/controller/appdefinition/networkpolicy.go

```diff
@@ -0,0 +1,190 @@
+package appdefinition
+
+import (
+	"fmt"
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/acorn/pkg/labels"
+	"github.com/acorn-io/baaah/pkg/router"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"strconv"
+	"strings"
+)
+
+func NetworkPolicy(req router.Request, resp router.Response) error {
+	cfg, err := config.Get(req.Ctx, req.Client)
+	if err != nil {
+		return err
+	} else if *cfg.DisableNetworkPolicies {
+		return nil
+	}
+
+	app := req.Object.(*v1.AppInstance)
+	appNamespace := app.ObjectMeta.Namespace // this is where the AppInstance lives
+	podNamespace := app.Status.Namespace     // this is where the app is actually running
+
+	// first, create the NetworkPolicy for the whole app
+	resp.Objects(&networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      app.Name,
+			Namespace: podNamespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: labels.Managed(app),
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{{
+				From: []networkingv1.NetworkPolicyPeer{{
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							labels.AcornAppNamespace: appNamespace,
+						},
+					}},
+				},
+			}},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		},
+	})
+
+	// get needed configuration information
+	conf, err := config.Get(req.Ctx, req.Client)
+	if err != nil {
+		return err
+	}
+	var ingressNamespace, podCIDR string
+	if conf.IngressControllerNamespace != nil {
+		ingressNamespace = *conf.IngressControllerNamespace
+	} else {
+		ingressNamespace = ""
+	}
+	if conf.PodCIDR != nil && *conf.PodCIDR != "" {
+		podCIDR = *conf.PodCIDR
+	} else {
+		podCIDR = "0.0.0.0/0"
+	}
+
+	// next, create NetworkPolicies for each container in the app that has a published port
+	// these policies allow ingress from anywhere
+	for containerName, container := range app.Status.AppSpec.Containers {
```

Comment:
Do we need to do this for `app.Status.AppSpec.Jobs` too?

Reply:
Yep we definitely do. Good call.